pages/science/reactors/cvodes.rst

```diff
@@ -0,0 +1,244 @@
+``ReactorNet::initialize()`` routine. ``CVODES`` requires that its user provide a C function that defines their ODE, 
+able to compute the right-hand side of the ODE system (dy/dt) for a given value of the independent variable, `t`, 
+and the state vector, ``y``. For more information about ODE right-hand side function requirements, 
+see `CVODES User Guide <https://sundials.readthedocs.io/en/latest/cvodes/Usage/SIM.html#user-supplied-functions>`__.
+
+The ``CVodesIntegrator`` wrapper class provides a useful C++ interface for configuring this C function by pairing with 
+``FuncEval``, an abstract base class for ODE right-hand-side function evaluators (`Documentation 
+<https://cantera.org/documentation/docs-2.6/doxygen/html/d1/dd1/classCantera_1_1FuncEval.html>`__). Classes derived 
+from ``FuncEval`` will implement the evaluation of the provided ODE system.
+
+An ODE right-hand-side evaluator is always needed in the ODE solution process (it's the only way to describe the system!), and for that reason a `FuncEval` object is a required parameter 
+when initializing any type of ``Integrator``.
+
+Let's take a look at how ``ReactorNet`` implements this ``FuncEval`` object. ``ReactorNet`` actually points to itself when 
+defining a ``FuncEval`` type, meaning it defines *itself* as a ``FuncEval`` derivative.
+
+Then, ``ReactorNet`` initializes the ``Integrator``, using a reference to itself (as a ``FuncEval``) from the 
+`this <https://en.cppreference.com/w/cpp/language/this>`__ pointer.
+
+To be a valid ``FuncEval`` object, a ``ReactorNet`` needs to provide implementations for all of ``FuncEval``'s 
+virtual functions, particularly the actual ODE right-hand-side computation 
+function, ``FuncEval::eval()``. Note that this is declared as a `pure virtual 
+<https://en.cppreference.com/w/cpp/language/abstract_class>`__ function, which makes 
+``FuncEval`` an abstract class:
+
+To evaluate the reactor governing equations the following parameters must be known:
+
+@param[in] t time.
+@param[out] LHS pointer to start of vector of left-hand side 
+coefficients for governing equations, length m_nv, default values 1
+@param[out] RHS pointer to start of vector of right-hand side 
+coefficients for governing equations, length m_nv, default values 0
+    
+.. code-block::
+    virtual void eval(double t, double* LHS, double* RHS);
+
+``eval()`` is called by ``ReactorNet::eval``.
+
+The above code shows the necessary inputs for solving the ODEs using the ``eval()`` function. ``eval()`` takes in the
+value of each state variable derivative (``ydot``) at a time ``t``, and will write the integrated values for each
+state varaible to the solution vector (``y``).
+
+Step 7: ``eval()`` is called to solve provided ODEs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Along with the rest of ``FuncEval``'s virtual functions, an appropriate override is provided for ``FuncEval::eval()`` in 
+``ReactorNet``
+
+.. code-block::
+
```

Comment:
You should tell this that it's C++ code so the following code will be formatted nicely.